### PR TITLE
docs: Add Getting Started guide for Gateway API support

### PR DESCRIPTION
**Pull request by sayboras**
Relates: https://github.com/cilium/cilium/pull/21749

The changes can be reviewed in https://deploy-preview-21908--docs-cilium-io.netlify.app/

```release-note
docs: Add Getting Started guide for Gateway API support
```

### CODE REVIEW

File: Documentation/network/servicemesh/gateway-api/gateway-api.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    Even though Cilium only supports above resources, the current
+    implementation passes all the conformance tests in the Gateway
+    API suite.
+
+.. Note::
+
+    Similar to Ingress, Gateway API controller creates a service of
+    LoadBalancer type, so your environment will need to support this.
+
```
Comment (qmonnet) [inline Documentation/network/servicemesh/gateway-api/gateway-api.rst]: This should probably be in the prerequisites instead

File: Documentation/network/servicemesh/gateway-api/gateway-api.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- `GatewayClass <https://gateway-api.sigs.k8s.io/api-types/gatewayclass/>`_
+- `Gateway <https://gateway-api.sigs.k8s.io/api-types/gateway/>`_
+- `HTTPRoute <https://gateway-api.sigs.k8s.io/api-types/httproute/>`_
+- `ReferenceGrant <https://gateway-api.sigs.k8s.io/api-types/referencegrant/>`_
+
+.. Note::
+
+    Even though Cilium only supports above resources, the current
+    implementation passes all the conformance tests in the Gateway
+    API suite.
```
Comment (qmonnet) [inline Documentation/network/servicemesh/gateway-api/gateway-api.rst]: Not clear to me what this means (but I'm not familiar with the concepts here, in particular the Gateway API suite).

Comment (sayboras) [reply Documentation/network/servicemesh/gateway-api/gateway-api.rst]: yeah, this is a little bit confusing, so it's better just remove it.

File: Documentation/network/servicemesh/gateway-api/gateway-api.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    https://docs.cilium.io
+
+.. _gs_gateway_api:
+
+*******************
+Gateway API Support
+*******************
+
+Cilium supports Gateway API v0.5.1 for below resources, all the upstream conformance tests
+are passed.
```
Comment (qmonnet) [inline Documentation/network/servicemesh/gateway-api/gateway-api.rst]: Nit: Maybe clarify this, too?

```suggestion
Cilium supports Gateway API v0.5.1 for the resources listed below, and passes `all upstream conformance tests <add_a_link_here>`__.
```

(Or `all related upstream conformance tests` if they're just the tests relating to supported resources?)

**Review by qmonnet — APPROVED** [review]
Thanks!

File: Documentation/network/servicemesh/gateway-api/gateway-api.rst (hunk 3 above)
Comment (youngnick) [reply Documentation/network/servicemesh/gateway-api/gateway-api.rst]: I think that `all the Core conformance tests, plus the ReferenceGrant extended tests` is the most correct. (This should change in the next release, when ReferenceGrant is included in the core suite).

**Review by qmonnet — APPROVED** [review]

**Comment by qmonnet** [comment]
I'd label with `release-note/misc`? There's no new feature introduced in this PR, only documentation?

**Comment by nvibert** [comment]
As mentioned on Slack, I think the docs missed the part where we need to deploy the GatewayClass. I may have missed something though but if not, we just need to add a GatewayClass Resource for Cilium on this manifest: https://github.com/cilium/cilium/blob/master/examples/kubernetes/gateway/basic-http.yaml

**Comment by sayboras** [comment]
> As mentioned on Slack, I think the docs missed the part where we need to deploy the GatewayClass. I may have missed something though but if not, we just need to add a GatewayClass Resource for Cilium on this manifest: https://github.com/cilium/cilium/blob/master/examples/kubernetes/gateway/basic-http.yaml

Default GatewayClass is installed as part of cilium helm chart, provided that CRD is installed in advance.

**Review by youngnick — APPROVED** [review]
LGTM, although we could possibly emphasize the requirement to preinstall the CRDs for this experimental version.

(The install order is tricky because Gateway API also has a validating webhook for objects, that really should be installed at some point, but we run the risk of circular dependencies there).

**Pull request merged**